sdk/ml/azure-ai-ml/azure/ai/ml/entities/_schedule/schedule.py

```diff
@@ -91,6 +88,84 @@ def provisioning_state(self):
         """
         return self._provisioning_state
 
+    def dump(self, dest: Union[str, PathLike, IO[AnyStr]], **kwargs) -> None:
+        """Dump the schedule content into a file in yaml format.
+
+        :param dest: The destination to receive this schedule's content.
+            Must be either a path to a local file, or an already-open file stream.
+            If dest is a file path, a new file will be created,
+            and an exception is raised if the file exists.
+            If dest is an open file, the file will be written to directly,
+            and an exception will be raised if the file is not writable.
+        :type dest: Union[str, PathLike, IO[AnyStr]]
+        """
+        path = kwargs.pop("path", None)
+        yaml_serialized = self._to_dict()
+        dump_yaml_to_file(dest, yaml_serialized, default_flow_style=False, path=path, **kwargs)
+
+    @classmethod
+    def _get_validation_error_target(cls) -> ErrorTarget:
+        return ErrorTarget.SCHEDULE
+
+    @classmethod
+    def _resolve_cls_and_type(cls, data, params_override):
+
+        if "create_job" in data:
```

Comment:
This seems hacky, do we have other ways to determine if a schedule is job, import data or model monitoring?